shell/platform/darwin/ios/framework/Source/FlutterTextInputPlugin.mm

```diff
@@ -2340,28 +2342,32 @@ - (void)handleMethodCall:(FlutterMethodCall*)call result:(FlutterResult)result {
 }
 
 - (void)handlePointerUp:(CGFloat)pointerY {
-  // View must be loaded at this point.
-  UIScreen* screen = _viewController.flutterScreenIfViewLoaded;
-  CGFloat screenHeight = screen.bounds.size.height;
-  CGFloat keyboardHeight = _keyboardRect.size.height;
-  BOOL shouldDismissKeyboard = (screenHeight - (keyboardHeight / 2)) < pointerY;
-  [UIView animateWithDuration:0.3f
-      animations:^{
-        double keyboardDestination =
-            shouldDismissKeyboard ? screenHeight : screenHeight - keyboardHeight;
-        _keyboardViewContainer.frame = CGRectMake(
-            0, keyboardDestination, _viewController.flutterScreenIfViewLoaded.bounds.size.width,
-            _keyboardViewContainer.frame.size.height);
-      }
-      completion:^(BOOL finished) {
-        if (shouldDismissKeyboard) {
-          [self.textInputDelegate flutterTextInputView:self.activeView
-              didResignFirstResponderWithTextInputClient:self.activeView.textInputClient];
-          [self dismissKeyboardScreenshot];
-        } else {
-          [self showKeyboardAndRemoveScreenshot];
+  if (_keyboardView.superview != nil) {
+    // Done to avoid the issue of a pointer up done without a screenshot
+    //  View must be loaded at this point.
+    UIScreen* screen = _viewController.flutterScreenIfViewLoaded;
+    CGFloat screenHeight = screen.bounds.size.height;
+    CGFloat keyboardHeight = _keyboardRect.size.height;
+    // Negative velocity indicates a downward movement
+    BOOL shouldDismissKeyboardBasedOnVelocity = _pointerYVelocity < 0;
+    [UIView animateWithDuration:0.3f
```

Comment:
can you make it a constant at the top of file?

Reply:
Definitely, added a description as well.